ambry-api/src/test/java/com.github.ambry/router/InMemoryRouter.java

```diff
@@ -102,6 +103,23 @@ public BlobProperties getBlobProperties() {
     public ByteBuffer getBlob() {
       return ByteBuffer.wrap(blob.array());
     }
+
+    /**
+     * @param range the {@link ByteRange} for the blob, or null.
+     * @return the blob content within the provided range, or the entire blob, if the range is null.
+     */
+    public ByteBuffer getBlobRange(ByteRange range) {
+//      ByteBuffer blobBuffer = getBlob();
+      if (range == null) {
+        return getBlob();
+      }
+      ByteRange resolvedRange = range.toResolvedByteRange(blob.array().length);
+      return ByteBuffer.wrap(Arrays
```

Comment:
Looks like you have done something in the commented code except that you tried to set the position instead of starting at the particular array offset. Is that possible?